tsl/src/bgw_policy/compression_api.c

```diff
@@ -34,6 +34,11 @@
 #define DEFAULT_RETRY_PERIOD                                                                       \
 	DatumGetIntervalP(DirectFunctionCall3(interval_in, CStringGetDatum("1 hour"), InvalidOid, -1))
 
+/* Default max schedule period for the compression policy is 12 hours. The actual schedule period
+ * will be chunk_interval/2 if the chunk_interval is < 12 hours. */
+#define DEFAULT_MAX_SCHEDULE_PERIOD                                                                \
+	DatumGetIntervalP(DirectFunctionCall3(interval_in, CStringGetDatum("12 hours"), InvalidOid, -1))
+
```

Comment:
This looks a little too much work for a constant. Maybe uint64 12 * 3600 * 1000 * 1000 instead?

Reply:
I have no strong opinion on this. Defining the const in that way was my initial plan, but the other costs are also defined in the more complicated way. So, I decided to be consistent with the other consts in this file.

Reply:
Not a good practice I think, when you use a const define you don't expect it to do a memory allocation inside (`interval_in` does). And they are compared to int values, so no point to hide the internal representation.

Reply:
Changed